### PR TITLE
CIV-7618 CCD to CUI Common Part Admit

### PR DESCRIPTION
**Pull request by kenneth-hmcts**
### JIRA link ###
https://tools.hmcts.net/jira/browse/CIV-7618

### Change description ###
This PR contains changes for CCD to CUI translation

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/common/models/civilClaimResponse.ts [hunk 1]
```diff
@@ -135,6 +141,12 @@ export interface CCDClaim extends ClaimUpdate {
   respondent1DQRecurringExpenses?: CCDRecurringExpenses[];
   respondent1DQRecurringExpensesFA?: CCDRecurringExpenses[];
   respondent1LiPResponse?: CCDRespondentLiPResponse;
+  respondent1DQLanguage?: CCDWelshLanguageRequirements;
```
Comment (laarmada) [inline src/main/common/models/civilClaimResponse.ts:144]: only to take care of that the size of this number of vbles is near to be out of control and unmanageable... could be group by , for example in that case DQ??

File: src/main/services/translation/convertToCUI/convertToCUIHearing.ts [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+      };
+      hearing.cantAttendHearingInNext12Months = toCUIGenericYesNo(ccdClaim.respondent1DQHearingSmallClaim.unavailableDatesRequired);
+    }
+    if(ccdClaim.respondent1DQHearingFastClaim) {
+      hearing.unavailableDatesForHearing = {
+        items : toCUIUnavailableDates(ccdClaim.respondent1DQHearingFastClaim.unavailableDates),
+      };
+      hearing.cantAttendHearingInNext12Months = toCUIGenericYesNo(ccdClaim.respondent1DQHearingFastClaim.unavailableDatesRequired);
+    }
+    if (ccdClaim.respondent1LiPResponse?.respondent1DQExtraDetails?.whyUnavailableForHearing) {
```
Comment (laarmada) [inline src/main/services/translation/convertToCUI/convertToCUIHearing.ts:29]: that is not the same as:
hearing.whyUnavailableForHearing.reason = ccdClaim.respondent1LiPResponse.respondent1DQExtraDetails.whyUnavailableForHearing ? ccdClaim.respondent1LiPResponse?.respondent1DQExtraDetails?.whyUnavailableForHearing : ....;

Comment (laarmada) [reply src/main/services/translation/convertToCUI/convertToCUIHearing.ts:29]: the same for each on if...

File: src/main/services/translation/convertToCUI/convertToCUIHearing.ts [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+    return hearing;
+  }
+};
+
+export const toCUISpecificCourtLocation = (specificCourtLocation: CCDSpecificCourtLocations) : SpecificCourtLocation=> {
+  return new SpecificCourtLocation(toCUIYesNo(specificCourtLocation.requestHearingAtSpecificCourt),specificCourtLocation.caseLocation?.baseLocation,specificCourtLocation.reasonForHearingAtSpecificCourt);
+};
+
+function toCUIUnavailableDates(ccdUnavailableDates: CCDUnavailableDates[]) : UnavailableDatePeriod[] {
+  if (!ccdUnavailableDates?.length) return undefined;
```
Comment (laarmada) [inline src/main/services/translation/convertToCUI/convertToCUIHearing.ts:55]: avoid return undefined for lenght reason you can control the return in the place which you call that function

File: src/main/services/translation/convertToCUI/convertToCUIVulnerability.ts [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import {VulnerabilityQuestions} from 'models/directionsQuestionnaire/vulnerabilityQuestions/vulnerabilityQuestions';
+import {CCDVulnerability} from 'models/ccdResponse/ccdVulnerability';
+import {toCUIYesNo} from 'services/translation/convertToCUI/convertToCUIYesNo';
+
+export const toCUIVulnerability = (ccdVulnerability: CCDVulnerability) : VulnerabilityQuestions => {
+  if(ccdVulnerability){
+    const vulnerabilityQuestions: VulnerabilityQuestions = new VulnerabilityQuestions();
```
Comment (laarmada) [inline src/main/services/translation/convertToCUI/convertToCUIVulnerability.ts:7]: here you are mixing the "new" 'OOD' with the json object structure:
vulnerabilityQuestions.vulnerability = {
      option: toCUIYesNo(ccdVulnerability.vulnerabilityAdjustmentsRequired),
      vulnerabilityDetails: ccdVulnerability.vulnerabilityAdjustments,
    };
... I don't like mix it. Only a comment

File: src/main/services/translation/convertToCUI/convertToCUIWitnesses.ts [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+          firstName: ccdWitnessDetails.value?.firstName,
+          lastName: ccdWitnessDetails.value?.lastName,
+          email: ccdWitnessDetails.value?.emailAddress,
+          telephone: ccdWitnessDetails.value?.phoneNumber,
+          details: ccdWitnessDetails.value?.reasonForWitness,
+        };
+      });
+    otherWitness.option = toCUIYesNo(ccdWitnesses.witnessesToAppear);
+    otherWitness.witnessItems = witnessItems;
+    const witness : Witnesses = {
```
Comment (laarmada) [inline src/main/services/translation/convertToCUI/convertToCUIWitnesses.ts:21]: that is the same as:
return new Witnesses (otherWitness);

**Review by laarmada — CHANGES_REQUESTED** [review]

**Review by neeta-hmcts — APPROVED** [review]

**Pull request merged**